lib/seahorn/BvOpSem2Allocators.cc

```diff
@@ -303,19 +303,22 @@ class StaticOpSemAllocator : public OpSemAllocator {
       auto rangeValCrab = m_sem.getCrabInstRng(*operand_inst);
       auto rangeValLVI = m_sem.getLVIInstRng(*operand_inst);
       uint64_t ub_crab = rangeValCrab.getUpper().getLimitedValue();
+      if (rangeValCrab.isFullSet() || rangeValCrab.isEmptySet()) {
```

Comment:
This code is sound but maybe too conservative. If a range is empty then it means that the instruction is unreachable so the check is actually safe (see comments here https://github.com/seahorn/clam/blob/master/include/clam/ClamQueryAPI.hh#L27).

Reply:
Should be inserted a call for `llvm_unreachable` if the set is empty?

Reply:
No. My point was that it is perfectly fine that a memory allocation is unreachable and clam can detect that by returning an empty interval. `llvm_unreachable` has a different purpose. This function aborts and it's intended to be called when some code (at the llvm side) is executed but it shouldn't. The unreachability I'm talking about is in the analyzed code not in the SeaHorn code.

Reply:
I thought this code was related to `is_dereferenceable` so that's why having an empty interval it can avoid a check. But then I realized this is different. It's just for helping the BMC engine to reason about allocation sizes. So I would leave it like that. Sorry about the digression.

Reply:
Yes. This code is used for inferring allocation size. Thanks for telling me the `llvm_unreachable` thing.